### PR TITLE
Cleanup names package

### PR DESCRIPTION
**Pull request by iawia002**
Signed-off-by: Xinzhao Xu <z2d@jifangcheng.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Clean up the `names` package and modify the comments format.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @RainbowMango

### CODE REVIEW

File: pkg/util/names/names.go [hunk 1]
```diff
@@ -16,12 +16,10 @@ const (
 	// - kube-public
 	// - kube-node-lease
 	KubernetesReservedNSPrefix = "kube-"
-	//NamespaceKarmadaSystem is reserved namespace
+	// NamespaceKarmadaSystem is reserved namespace
 	NamespaceKarmadaSystem = "karmada-system"
-	//NamespaceKarmadaCluster is reserved namespace
+	// NamespaceKarmadaCluster is reserved namespace
```
Comment (RainbowMango) [inline pkg/util/names/names.go:21]: Thanks. I've noticed this issue too.

File: pkg/util/names/names.go [hunk 2]
```diff
@@ -16,12 +16,10 @@ const (
...
 	NamespaceKarmadaCluster = "karmada-cluster"
-	//NamespaceDefault is reserved namespace
-	NamespaceDefault = "default"
```
Comment (RainbowMango) [inline pkg/util/names/names.go:24]: The `const()` block used to define reserved namespaces,  people could get the whole list from here when they read code.
So, I don't think it's worth to remove `default` from here and replace by a reference.

Comment (iawia002) [reply pkg/util/names/names.go]: You're right, I hadn't thought about that.

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**